### ISSUE TITLE
Partition MainWindow class

### ISSUE THREAD

**Issue by WargodHernandez** [CloudCompare/CloudCompare#937]
I would like to see the MainWindow class have the user interface decoupled a bit from the operations it performs. Specifically to reduce mostly repetitive code in MainWindow, ccCommandLineCommands, and some Plugin functionalities.

for example 

```C++
void MainWindow::doActionSubsample()

CommandSubsample::CommandSubsample() : ccCommandLineInterface::Command("Subsample", COMMAND_SUBSAMPLE) {}

bool CommandSubsample::process(ccCommandLineInterface &cmd)

 ```

By having a data structure that represents the options usually selected in the SubSample dialog or the parameters passed to the commandline a more generic Subsample function could be created that the window and the commandline class both call. 

```C++
struct ParamStructSubSample
	{
        enum SubSampleMethod {RANDOM = 0, SPACE = 1, OCTREE = 2};
        SubSampleMethod method;
        double samplingValue;
        bool useSF;
        double sfMinSpacing;
        double sfMaxSpacing;
        }

CCLib::ReferenceCloud* ccSubSample(ccGenericPointCloud* cloud, ParamStructSubSample params, CCLib::GenericProgressCallback* progressCb)

 ```

I think this will help in the long run getting as many command line commands as possible and help with scripting plugins, and possibly reduce the size of some of the monster files like mainwindow.cpp which is currently 10524 lines long or ccCommandLineCommands.cpp 4083 lines

The SubSample example is just that, it isn't a specific request for SubSample.

**Comment by dgirardeau** [comment]
Good idea. This was also partly started by Andy when he created the 'ccEntityAction' header file.

You have my blessings to continue in this direction ;)

**Comment by WargodHernandez** [comment]
Great Thanks, I'll try to make reasonable PR's instead of changing everything all at once :)
